Source/CompilationDVDView.cpp

```diff
@@ -0,0 +1,341 @@
+			.End();
+}
+
+
+CompilationDVDView::~CompilationDVDView()
+{
+	delete fBurnerThread;
+}
+
+
+#pragma mark -- BView Overrides --
+
+
+void
+CompilationDVDView::AttachedToWindow()
+{
+	BView::AttachedToWindow();
+
+	fDVDButton->SetTarget(this);
+	fDVDButton->SetEnabled(true);
+	
+	fImageButton->SetTarget(this);
+	fImageButton->SetEnabled(false);
+
+	fBurnButton->SetTarget(this);
+	fBurnButton->SetEnabled(false);
+}
+
+
+void
+CompilationDVDView::MessageReceived(BMessage* message)
+{
+	switch (message->what) {
+		case kNoPathMessage:
+		case kChooseDirectoryMessage:
+			_ChooseDirectory();
+			break;
+		case kBurnDiscMessage:
+			BurnDisc();
+			break;
+		case kBuildImageMessage:
+			BuildISO();
+			break;
+		case B_REFS_RECEIVED:
+			_OpenDirectory(message);
+			break;
+		case kBurnerMessage:
+			_BurnerOutput(message);
+			break;
+		default:
+			BView::MessageReceived(message);
+	}
+}
+
+
+#pragma mark -- Private Methods --
+
+
+bool
+DVDRefFilter::Filter(const entry_ref* ref, BNode* node,
+	struct stat_beos* stat, const char* filetype)
+{
+	if (S_ISDIR(stat->st_mode))
+		return true;
+
+	if (S_ISLNK(stat->st_mode)) {
+		// Traverse symlinks
+		BEntry entry(ref, true);
+		return entry.IsDirectory();
+	}
+
+	return false;
+}
+
+
+void
+CompilationDVDView::_ChooseDirectory()
+{
+	if (fOpenPanel == NULL) {
+		fOpenPanel = new BFilePanel(B_OPEN_PANEL, new BMessenger(this), NULL,
+			B_DIRECTORY_NODE, false, NULL, new DVDRefFilter(), true);
+	}
+	fOpenPanel->Show();
+}
+
+
+void
+CompilationDVDView::_OpenDirectory(BMessage* message)
+{
+	entry_ref ref;
+	if (message->FindRef("refs", &ref) != B_OK)
+		return;
+
+	BEntry entry(&ref, true);	// also accept symlinks
+	BNode node(&entry);
+	if ((node.InitCheck() != B_OK) || !node.IsDirectory())
+		return;
+
+	fDirPath->SetTo(&entry);
+	if (strcmp("VIDEO_TS", fDirPath->Leaf()) == 0)
+		fDirPath->GetParent(fDirPath);
+	else {
+		BPath testPath(fDirPath->Path());
+		testPath.Append("VIDEO_TS");
+		get_ref_for_path(testPath.Path(), &ref);
+		if (!BEntry(&ref).Exists()) {
+			fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+				"Didn't find a VIDEO_TS folder", "Status notification"));
+			return;
+		}
+	}
+
+	fPathView->SetText(fDirPath->Path());
+	fImageButton->SetEnabled(true);
+	fBurnButton->SetEnabled(false);
+	fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT("Build the DVD UDF image",
+		"Status notification"));
+
+	BPath audioTSPath(fDirPath->Path());
+	audioTSPath.Append("AUDIO_TS");
+	create_directory(audioTSPath.Path(), 0777);
+}
+
+
+void
+CompilationDVDView::_BurnerOutput(BMessage* message)
+{
+	BString DVD;
+
+	if (message->FindString("line", &DVD) == B_OK) {
+		DVD << "\n";
+		fBurnerInfoTextView->Insert(DVD.String());
+		fBurnerInfoTextView->ScrollTo(0.0, 2000.0);
+	}
+	int32 code = -1;
+	if ((message->FindInt32("thread_exit", &code) == B_OK) && (step == 1)) {
+		BString infoText(fBurnerInfoTextView->Text());
+		if (infoText.FindFirst(
+			"mkisofs: Unable to make a DVD-Video image.\n") != B_ERROR) {
```

Comment:
I hope mkisofs never gets localized with system global settings, otherwise the string might be another. Is the "code" not 0 for errors?

Reply:
I hope that, too. :)
Unfortunately "code" is always 0. See the [Todo](https://github.com/HaikuArchives/BurnItNow/blob/master/Source/CommandThread.cpp#L205): "TODO adjust this based on the actual command exit code".
Unfortunately, I have no idea how to implement that. Patch welcome! :)

Reply:
Yeah it seems missing in https://github.com/HaikuArchives/BurnItNow/blob/master/Source/CommandThread.cpp#L185

You could try to move exitval to a member of the class CommandThread, then uses it for https://github.com/HaikuArchives/BurnItNow/blob/master/Source/CommandThread.cpp#L206

Reply:
I tried using a member variable, but get the error:
```
 In function `static int32 CommandThread::_Thread(void *)':
 invalid use of member `CommandThread::fexitVal' in static member function
```
If I remove the static from the functions I get errors, and if make the new member variable static, I get others...

I have to admit, I'm not that familiar with "static" and its consequences...

In any case, I loaded the original code in Debugger and it seems the code that sets the exitval inside `if (pipe.ReadLines(stdOutAndErrPipe, reader) != B_OK)` is never executed, even when mkisofs fails.